app/javascript/crayons/Modal/Modal.jsx

```diff
@@ -17,74 +17,66 @@ function getAdditionalClassNames({ size, className }) {
   return additionalClassNames;
 }
 
-export class Modal extends Component {
-  state = {
-    visible: true,
-  };
+export const Modal = ({
+  children,
+  size = 'default',
+  className,
+  title,
+  overlay,
+  onClose,
+}) => {
+  const Icon = () => (
+    <svg
+      width="24"
+      height="24"
+      viewBox="0 0 24 24"
+      className="crayons-icon"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636l4.95 4.95z" />
+    </svg>
+  );
 
-  render() {
-    const { visible } = this.state;
-    const {
-      children,
-      size = 'default',
-      className,
-      title,
-      overlay,
-    } = this.props;
-
-    const Icon = () => (
-      <svg
-        width="24"
-        height="24"
-        viewBox="0 0 24 24"
-        className="crayons-icon"
-        xmlns="http://www.w3.org/2000/svg"
-      >
-        <path d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636l4.95 4.95z" />
-      </svg>
-    );
-
-    return visible ? (
-      <div
-        className={`crayons-modal${getAdditionalClassNames({
-          size,
-          className,
-        })}`}
-      >
-        <div className="crayons-modal__box">
-          {title.length > 0 && title && (
-            <div className="crayons-modal__box__header">
-              <h2>{title}</h2>
-              <Button
-                icon={Icon}
-                variant="ghost"
-                contentType="icon"
-                title="Close"
-                onClick={(_event) => {
-                  this.setState({ visible: !visible });
-                }}
-              />
-            </div>
-          )}
-          <div className="crayons-modal__box__body">{children}</div>
-        </div>
-        {overlay && <div className="crayons-modal__overlay" />}
+  return (
+    <div
+      className={`crayons-modal${getAdditionalClassNames({
+        size,
+        className,
+      })}`}
+    >
+      <div className="crayons-modal__box">
+        {title.length > 0 && title && (
+          <div className="crayons-modal__box__header">
+            <h2>{title}</h2>
+            <Button
+              icon={Icon}
+              variant="ghost"
+              contentType="icon"
+              title="Close"
+              onClick={onClose}
+            />
+          </div>
+        )}
+        <div className="crayons-modal__box__body">{children}</div>
```

Comment:
There's an awkward side-scroll in this box since we have some contents which can be slightly wider than 100% I think. `overflow-x: hidden` would be a quick fix which would work for now if we don't want to immediately rewrite some of the inner functionality.

Reply:
i'll look into this.